### PR TITLE
Add a page to explain 'shared components' and where to find them

### PR DESCRIPTION
**Pull request by mperrotti**
Closes https://github.com/github/primer/issues/1915

Unfortunately the `ui/packages/` component metadata has been de-prioritized. However, we can still point Hubbers towards components that are meant to shared but aren't in Primer yet.

### CODE REVIEW

**Comment by emilybrick** [comment]
Something about putting this within the list of components makes it hard to discover for me, especially since it's the only item in that list that's not like the others (not a component page). 

What if we instead had a section for `GitHub staff` or 'Internal' that we kept specific documentation behind? Even if it's just shared components for now 🤔 , I suspect this list may grow. 
Screenshot 2023-06-05 at 12 12 44 PM

File: content/components/privately-shared-components.mdx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: Privately shared components
```
Comment (tallys) [inline content/components/privately-shared-components.mdx:2]: I think we should discuss this with @emilybrick and run this through a "what misunderstanding that can occur" if we publish this on our public page and make sure this is in alignment with the rest of our docs strategy. Also, "Private" and "Shared" are opposites, perhaps we can find a name that provides more clarity.

**Review by tallys — CHANGES_REQUESTED** [review]

File: content/components/privately-shared-components.mdx (hunk 1 above)
Comment (emilybrick) [reply content/components/privately-shared-components.mdx:2]: I'm good with `GitHub shared components` fwiw.

File: content/components/github-shared-components.mdx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+title: GitHub shared components
+description: Components that are shared by GitHub feature teams, but are not publicly available in Primer yet.
+---
+
+<Note>This information is only relevant to GitHub staff.</Note>
+
+These components support new design patterns that have been piloted by feature teams outside of Primer. While we encourage relying only on Primer components, we also value exploring new ideas without exhaustive the vetting process required for inclusion in Primer.
```
Comment (emilybrick) [inline content/components/github-shared-components.mdx:8]:
```suggestion
These components support new design patterns that have been piloted by feature teams outside of Primer. While we encourage relying only on Primer components, we also value exploring new ideas without exhausting the vetting process required for inclusion in Primer.
```

File: content/components/github-shared-components.mdx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+Should a design pattern from the `ui/packages/` monorepo gain widespread adoption, it becomes a strong candidate for upstreaming into Primer. If your component is data-dependent or feature-specific, our Primer maintainers can help evolve it into a more versatile pattern suitable for multiple use-cases.
+
+For more information on the upstreaming process and how existing components can be included in Primer, refer to our documentation on [upstreaming existing components](/guides/contribute/adding-new-components#upstreaming-to-primer).
+
+## Finding these components
+
+GitHub shared components are documented in a private Storybook instance: [ui/packages/ Storybook](https://gh.io/storybook) (only available to GitHub staff).
+
+Please note that these components will not exist in our Figma component library, [Primer Web](https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?type=design&node-id=1406%3A0&mode=design&t=Qi8hXoRhKLLrDhgf-1) (only available to GitHub staff), until they have been successfully upstreamed into Primer.
```
Comment (emilybrick) [inline content/components/github-shared-components.mdx:24]:
```suggestion
Please note that these components will not exist in our Figma component library, [Primer Web](https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?type=design&node-id=1406%3A0&mode=design&t=Qi8hXoRhKLLrDhgf-1), until they have been successfully upstreamed into Primer.
```

don't think you need that bit

**Review by emilybrick — APPROVED** [review]

**Review by emilybrick — APPROVED** [review]

File: content/components/github-shared-components.mdx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+title: GitHub shared components
+description: Components that are shared by GitHub feature teams, but are not publicly available in Primer yet.
+---
+
+<Note>This information is only relevant to GitHub staff.</Note>
+
+These components support new design patterns that have been piloted by feature teams outside of Primer. While we encourage relying only on Primer components, we also value exploring new ideas without the exhaustive vetting process required for inclusion in Primer.
```
Comment (tallys) [inline content/components/github-shared-components.mdx:8]:
```suggestion
These components are shared design patterns by GitHub feature teams developing application-specific components. We encourage relying only on Primer components where possible, and not all patterns will be upstreamed to Primer. 

```

Comment (tallys) [reply content/components/github-shared-components.mdx:8]: Hi @emilybrick and @mperrotti — minor edit to avoid confusion but I'd like to make sure that the:
1. Perception of our processes is respected (language to move away from exhausting/exhaustive toward more stable, trusted, resilient terms) 
2. Clear as day understanding that this is for GitHub application-level patterns that can, but may not be, upstreamed to Primer. It is not a draft state or experimental state. We are not "obscuring" any patterns or components from our Open Source library.

TLDR: can @mperrotti or @emilybrick run this through comparing against the language here? https://thehub.github.com/epd/engineering/dev-practicals/frontend/common-components/

Comment (mperrotti) [reply content/components/github-shared-components.mdx]: We were already pretty aligned with that post, but I just pushed up a few changes to make the alignment even stronger 💪

File: content/components/github-shared-components.mdx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+They're shared with other teams by developing them as React components within the `ui/packages/` monorepo. This monorepo, along with the larger monolith, provides a solid foundation with baseline configurations for linting, accessibility scanning using Axe, and Storybook previews.
+
+Engineers building UI components should refer to the guidance on The Hub: [Building reusable UI Components](https://thehub.github.com/epd/engineering/dev-practicals/frontend/common-components/) (only available to GitHub staff).
+
+## Finding these components
+
+GitHub shared components are documented in a private Storybook instance: [ui/packages/ Storybook](https://gh.io/storybook) (only available to GitHub staff).
+
+Please note that these components will not exist in our Figma component library, [Primer Web](https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?type=design&node-id=1406%3A0&mode=design&t=Qi8hXoRhKLLrDhgf-1), until they have been successfully upstreamed into Primer.
```
Comment (tallys) [inline content/components/github-shared-components.mdx:18]:
```suggestion
Please note that these components are **owned by the feature teams**, and will not exist in our Figma component library, [Primer Web](https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?type=design&node-id=1406%3A0&mode=design&t=Qi8hXoRhKLLrDhgf-1), unless they have been upstreamed into Primer.
```

**Review by emilybrick — APPROVED** [review]
Thank you for updating!
